File: app/src/main/java/com/example/ozakharc/redditclient/model/ItemCommentDao.java
package com.example.ozakharc.redditclient.model;


import android.arch.lifecycle.LiveData;
import android.arch.persistence.db.SimpleSQLiteQuery;
import android.arch.persistence.room.Dao;
import android.arch.persistence.room.Delete;
import android.arch.persistence.room.Insert;
import android.arch.persistence.room.OnConflictStrategy;
import android.arch.persistence.room.Query;

import java.util.List;

@Dao
public abstract class ItemCommentDao {
    @Insert(onConflict = OnConflictStrategy.REPLACE)
    public abstract void insert(ItemComment... items);

    @Query("SELECT * FROM itemcomment WHERE permalink=:permalink")
    public abstract LiveData<List<ItemComment>> search(final String permalink);

    @Query("DELETE FROM itemcomment")
    public abstract void deleteAll();
}

File: app/src/main/java/com/example/ozakharc/redditclient/model/ItemCommentDao.kt
//package com.example.ozakharc.redditclient.model
//
//import android.arch.lifecycle.LiveData
//import android.arch.persistence.room.Dao
//import android.arch.persistence.room.Insert
//import android.arch.persistence.room.OnConflictStrategy.REPLACE
//import android.arch.persistence.room.Query
//
//@Dao
//interface ItemCommentDao {
//
//    @Insert(onConflict = REPLACE)
//    fun insert(itemComment: ItemComment)
//
//    @Insert(onConflict = REPLACE)
//    fun insertAll(comments: List<ItemComment>)
//
//    @Query("DELETE from itemcomment")
//    fun deleteAll()
//
//    @Query("SELECT * from itemcomment WHERE permalink = :permalink")
//    fun search(permalink: String): LiveData<List<ItemComment>>
//
//}